### DISCUSSION TITLE
Wayland layer specific remaps?

### DISCUSSION THREAD

**Discussion by theoahfeldt** [xremap/xremap#624] [Ideas]
It is possible to configure application specific remaps that depend on the current active window. But wayland also has programs that do not run in windows but instead use [layers](https://wayland.app/protocols/wlr-layer-shell-unstable-v1) (e.g. app launchers). 

Personally I am on hyprland and I have an app launcher built in [AGS](https://github.com/Aylur/ags), and I am using the [emacs keymap](https://github.com/xremap/xremap/blob/master/example/emacs.yml) which specifies to apply to all applications but emacs. In this case, the keymap that gets used when I am using my app launcher depends on if emacs is the active window or not. That is, if I am working in emacs and launch my app launcher, the emacs keymap will not be active for the app launcher. If I am using any other application and launch my app launcher, the emacs keymap will be active for the app launcher.

It would be very convenient if it was possible to specify whether a keymap should apply to layers or not. I have searched the documentation and have not found any way of doing this, so if there is a way I would be very happy to know it. Otherwise, is it something that could be implemented?

**Comment by k0kubun** [comment]
I don't think we currently support such a concept. Feel free to open a PR that implements `layer:` filter just like `application:` and `window:`. I would merge such a PR.
